packages/dds/tree/src/util/utils.ts

```diff
@@ -563,3 +563,49 @@ export function defineLazyCachedProperty<
 	});
 	return obj as typeof obj & { [P in K]: V };
 }
+
+/**
+ * Copies a given property from one object to another if and only if the property is present.
+ * @param source - The object to copy the property from.
+ * If `source` is undefined or does not have the given property, then this function will do nothing.
+ * @param property - The property to copy.
+ * @param destination - The object to copy the property to.
+ * @remarks This function is useful for copying properties from one object to another when the property may not be present.
+ * If `property` is not present on `source`, then this function will leave `destination` untouched.
+ * This is different from doing `destination.foo = source.foo`, which will define a `"foo"` property on `destination` with the value `undefined`.
+ *
+ * If the type of `source` is known to have `property`, then this function asserts that the type of `destination` has `property` as well after the call.
+ *
+ * This function first reads the property value (if present) from `source` and then sets it on `destination`, as opposed to e.g. directly copying the property descriptor.
+ * @privateRemarks The first overload of this function allows auto-complete to suggest property names from `source`, but by having the second overload we still allow for arbitrary property names.
```

Comment:
Neat. I like it!

Reply:
Yea I was pleasantly surprised by how well this function turned out, typing-wise :)